sdk/src/main/scala/com/horizen/account/receipt/LogsBloom.scala

```diff
@@ -80,10 +80,14 @@ class LogsBloom() extends BytesSerializable {
 
   override def equals(obj: Any): Boolean = {
     obj match {
-      case l: LogsBloom => (this.bloomFilter diff l.getBloomFilter()).isEmpty
+      case l: LogsBloom => this.bloomFilter.sameElements(l.getBloomFilter())
       case _ => false
     }
   }
+
+  override def hashCode(): Int = {
+    31.hashCode() * this.bloomFilter.hashCode()
```

Comment:
I think `this.bloomFilter.hashCode()` is enough here.
why do we need `31.hashCode() *` part?

Reply:
I did these changes already in this PR: https://github.com/HorizenOfficial/Sidechains-SDK/pull/644/files#diff-deaa0e70e60ddbc5961d8b6188c566623deeea8110511599db00b24c0d5b1232R98
I used `util.Arrays.hashCode(filter)` as we want the hashCode to depend on the contents of the bitmask. I'm not sure if `filter.hashCode()` would do that too.